demo/demo.h

```diff
@@ -37,37 +44,207 @@ DECLARE_int32(demo_port);
+                        if (n) {
+                          response.Send(data.substr(0, n));
+                          data = data.substr(n);
+                        }
+                      }
+                    } catch (const std::exception& e) {
+                      std::cerr << "Exception in data serving thread: " << e.what() << std::endl;
+                    }
+                  },
+                  std::move(r)).detach();
+    });
+    HTTP(port).Register("/layout/meta", [](Request r) {
+      r.connection.SendHTTPResponse(ExampleMeta(),
+                                    "meta",
+                                    HTTPResponseCode::OK,
+                                    "application/json; charset=utf-8",
+                                    {{"Connection", "close"}, {"Access-Control-Allow-Origin", "*"}});
+    });
+    HTTP(port).Register("/layout", [](Request r) {
+      LayoutItem layout;
+      LayoutItem row;
+      layout.col.push_back(row);
+      r.connection.SendHTTPResponse(layout,
+                                    "layout",
+                                    HTTPResponseCode::OK,
+                                    "application/json; charset=utf-8",
+                                    {{"Connection", "close"}, {"Access-Control-Allow-Origin", "*"}});
+    });
+    // The "./static/" directory should be a symlink to "Web/build" or "Web/build-dev".
+    FileSystem::ScanDir("./static/", [&port](const std::string& filename) {
+      std::string filepath = "./static/" + filename;
+      std::string fileurl = "/static/" + filename;
+      // The default route points to the frontend entry point file.
+      if (filename == "index.html") {
+        fileurl.assign("/");
+      }
       // Hack with string ownership.
       // TODO(dkorolev): Make this cleaner.
-      std::string* ps = new std::string();
-      *ps = FileSystem::ReadFileAsString("static/" + fn);
-      HTTP(port).Register("/static/" + fn, [ps](Request r) {
-        r.connection.SendHTTPResponse(*ps, HTTPResponseCode::OK, "text/plain");
+      std::string* contentType = new std::string(DemoServer::GetFileMimeType(filename));
+      std::string* content = new std::string();
+      *content = FileSystem::ReadFileAsString(filepath);
+      HTTP(port).Register(fileurl, [content, contentType](Request r) {
+        r.connection.SendHTTPResponse(*content, HTTPResponseCode::OK, *contentType);
       });
     });
   }
 
   void Join() {
-    using bricks::net::api::HTTP;
-
     printf("Listening on port %d\n", port_);
     HTTP(port_).Join();
     printf("Done.\n");
   }
 
+ private:
   State state_;
   const int port_;
+
+  /**
+   * Gets the file extension from the file name or path.
+   */
+  static const std::string GetFileExtension(const std::string& filename) {
+    size_t fileextIndex = filename.find_last_of("/\\.");
+
+    if (fileextIndex == std::string::npos || filename[fileextIndex] != '.') {
```

Comment:
IIRC, the last part of this ```if```` is redundant, ```find_last_of``` guarantees that a non-```npos``` response means that the desired character was found.